### ISSUE TITLE
[CI] CompressibleBytesOutputStreamTests.testCompressionWithCallingMaterializeFails fails in 5.6 branch

### ISSUE THREAD

**Issue by dakrone** [elastic/elasticsearch#27648]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+periodic/951/console

```
==> Test Info: seed=878230952D17E4AB; jvm=1; suite=1
Suite: org.elasticsearch.transport.CompressibleBytesOutputStreamTests
  1> [2017-12-04T13:20:14,325][INFO ][o.e.t.CompressibleBytesOutputStreamTests] [testCompressionWithCallingMaterializeFails]: before test
  1> [2017-12-04T13:20:14,386][INFO ][o.e.t.CompressibleBytesOutputStreamTests] [testCompressionWithCallingMaterializeFails]: after test
  2> REPRODUCE WITH: gradle :core:test -Dtests.seed=878230952D17E4AB -Dtests.class=org.elasticsearch.transport.CompressibleBytesOutputStreamTests -Dtests.method="testCompressionWithCallingMaterializeFails" -Dtests.security.manager=true -Dtests.locale=ar-KW -Dtests.timezone\
=America/Recife
FAILURE 0.09s | CompressibleBytesOutputStreamTests.testCompressionWithCallingMaterializeFails <<< FAILURES!
   > Throwable #1: junit.framework.AssertionFailedError: Expected exception EOFException
   >    at __randomizedtesting.SeedInfo.seed([878230952D17E4AB:EB7A904C95C5C85D]:0)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2704)
   >    at org.elasticsearch.transport.CompressibleBytesOutputStreamTests.testCompressionWithCallingMaterializeFails(CompressibleBytesOutputStreamTests.java:94)
   >    at java.lang.Thread.run(Thread.java:748)
  2> NOTE: leaving temporary files on disk at: /home/hinmanm/es/elasticsearch/core/build/testrun/test/J0/temp/org.elasticsearch.transport.CompressibleBytesOutputStreamTests_878230952D17E4AB-001
  2> NOTE: test params are: codec=Asserting(Lucene62): {}, docValues:{}, maxPointsInLeafNode=159, maxMBSortInHeap=5.628844637258778, sim=RandomSimilarity(queryNorm=true,coord=crazy): {}, locale=ar-KW, timezone=America/Recife
  2> NOTE: Linux 4.13.12-300.fc27.x86_64 amd64/Oracle Corporation 1.8.0_151 (64-bit)/cpus=8,threads=1,free=438865776,total=514850816
  2> NOTE: All tests run in this JVM: [CompressibleBytesOutputStreamTests]
Completed [1/1] in 0.72s, 1 test, 1 failure <<< FAILURES!
```

This reproduces for me with:

```
gradle :core:test -Dtests.seed=878230952D17E4AB -Dtests.class=org.elasticsearch.transport.CompressibleBytesOutputStreamTests -Dtests.method="testCompressionWithCallingMaterializeFails" -Dtests.security.manager=true -Dtests.locale=ar-KW -Dtests.timezone=America/Recife
```

**Comment by dakrone** [comment]
Another one https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+multijob-unix-compatibility/os=opensuse/401/console

**Comment by jasontedor** [comment]
Closed by b98244e2869138e897689bb1f1056b3b85bcec0b
